packages/storage/src/implementation/list.ts

```diff
@@ -0,0 +1,110 @@
+/**
+ * @license
+ * Copyright 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Documentation for the listOptions and listResult format
+ */
+import { AuthWrapper } from './authwrapper';
+import { Location } from './location';
+import * as json from './json';
+import * as type from './type';
+import { ListResult } from '../list';
+
+/**
+ * Represents the simplified object metadata returned by List API.
+ * Other fields are filtered because list does not grant permission to get metadata.
+ */
+interface ListMetadataResponse {
+  name: string;
+  bucket: string;
+}
+
+/**
+ * Represents the JSON response of List API.
+ */
+interface ListResultResponse {
+  prefixes: string[];
+  items: ListMetadataResponse[];
+  nextPageToken?: string;
+}
+
+const MAXRESULTS_KEY = 'maxResults';
+const PAGETOKEN_KEY = 'pageToken';
+const PREFIXES_KEY = 'prefixes';
+const ITEMS_KEY = 'items';
+
+function fromBackendResponse(
```

Comment:
This is a much much clearer name (especially to someone that doesn't know the rest of the codebase (*)). Thanks!

(*) Like me.